docs/source/reader_and_writer.rst

```diff
@@ -172,6 +172,39 @@ Basically, it's just a high level api on top of the native
         'rewriteBatchedStatements': 'true',  # improves write throughput dramatically
     })
 
+.. _redis:
```

Comment:
Yes, we've decided we don't care about a reader currently since main usage of Redis is as a cache.

I've started implementing your key/value serializers suggestion as it does provide a more powerful api at the end of the day, but I stopped and I'm not sure it's a good idea anymore:
* It complicates code quite a lot.
* As a user I'd expect that `write_ext.redis` allows me to set similar parameters to the ones I can set through `write_ext.by_url`. That's true for all other datastores we implement here except for Kafka, for which we don't provide URL support at all.
* It becomes tedious from a user perspective to set simple serializing parameters like `compression` and `key_prefix` as now I have to define a function to do it.
* It provides more fine grained control than we will probably ever need. If the need arises internally or a user of the package requests it I'm all in for re-visiting.
* I've actually spent way more time on this PR than it was initially allocated so I can't invest much more.

So, unless you feel strongly about this, I'd prefer to stash it for a v2.